From 1a6d2f1fd045ba671153be922a033f4605185dfe Mon Sep 17 00:00:00 2001
From: Ming Tan <ming.tan@intel.com>
Date: Thu, 29 Aug 2019 10:41:16 +0800
Subject: [PATCH 3/6] Put the scan sequence of USB storage behand of general
 device.

The general device will not include USB storage.
In some platform, if the storage is set as SCSI, then it will be used as
general device in kernelflinger, and we need put it before USB storage.

Change-Id: Idec027ddcf67dcfba8cf5c4d8e95188f19833a1a
Tracked-On: OAM-85394
Signed-off-by: Ming Tan <ming.tan@intel.com>
---
 include/libkernelflinger/storage.h | 2 +-
 libkernelflinger/storage.c         | 4 ++--
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/include/libkernelflinger/storage.h b/include/libkernelflinger/storage.h
index 5965880..fb5a0cf 100755
--- a/include/libkernelflinger/storage.h
+++ b/include/libkernelflinger/storage.h
@@ -43,10 +43,10 @@ enum storage_type {
 	STORAGE_SATA,
 	STORAGE_NVME,
 	STORAGE_VIRTUAL,
+	STORAGE_GENERAL_BLOCK,
 #ifdef USB_STORAGE
 	STORAGE_USB,
 #endif
-	STORAGE_GENERAL_BLOCK,
 	STORAGE_ALL,
 };
 
diff --git a/libkernelflinger/storage.c b/libkernelflinger/storage.c
index 02f1bb3..fcf48d1 100644
--- a/libkernelflinger/storage.c
+++ b/libkernelflinger/storage.c
@@ -66,10 +66,10 @@ extern struct storage STORAGE(STORAGE_SDCARD);
 extern struct storage STORAGE(STORAGE_SATA);
 extern struct storage STORAGE(STORAGE_NVME);
 extern struct storage STORAGE(STORAGE_VIRTUAL);
+extern struct storage STORAGE(STORAGE_GENERAL_BLOCK);
 #ifdef USB_STORAGE
 extern struct storage STORAGE(STORAGE_USB);
 #endif
-extern struct storage STORAGE(STORAGE_GENERAL_BLOCK);
 
 
 static EFI_STATUS identify_storage(EFI_DEVICE_PATH *device_path,
@@ -85,10 +85,10 @@ static EFI_STATUS identify_storage(EFI_DEVICE_PATH *device_path,
 		, &STORAGE(STORAGE_SATA)
 		, &STORAGE(STORAGE_NVME)
 		, &STORAGE(STORAGE_VIRTUAL)
+		, &STORAGE(STORAGE_GENERAL_BLOCK)
 #ifdef USB_STORAGE
 		, &STORAGE(STORAGE_USB)
 #endif
-		, &STORAGE(STORAGE_GENERAL_BLOCK)
 	};
 
 	for (st = STORAGE_EMMC; st < STORAGE_ALL; st++) {
-- 
2.21.0

